crates/kornia-core-ops/src/error.rs

```diff
@@ -0,0 +1,9 @@
+use thiserror::Error;
+
+/// An error type for tensor operations.
+#[derive(Error, Debug, PartialEq)]
+pub enum TensorOpsError {
+    /// Dimension out of bounds.
+    #[error("Dimension out of bounds. The dimension {0} is out of bounds ({1}).")]
+    DimOutOfBounds(usize, usize),
```

Comment:
can we reuse or unify somehow with https://github.com/kornia/kornia-rs/blob/main/crates/kornia-core/src/tensor.rs#L23 ?

Reply:
I can add `DimOutOfBounds` to `TensorError`. Feels like it means something different to `IndexOutOfBounds`. The latter is when the index is greater than the size of the array and the former when the queried dimension/axis does not exist.

Reply:
I couldn't think of a scenario where this error is applicable outside of tensor operations (tensor indexing is already validated by the const generics) and therefore I created it in the core-ops crate initially.

Reply:
i see, yes makes sense. Make sure the description and error message is clear for both then